airflow/www/templates/airflow/dag.html

```diff
@@ -108,199 +108,221 @@ <h4 class="pull-right">
+                </label>
+              </span>
+              <span class="col-md-3">
+                <button id="btn_failed" type="submit" class="btn btn-primary btn-block">
+                  Mark Failed
+                </button>
+              </span>
+            </div>
+          </form>
+          <hr/>
+          <form method="POST" data-action="{{ url_for('Airflow.success') }}">
+            <input name="csrf_token" type="hidden" value="{{ csrf_token() }}"/>
+            <input name="dag_id" type="hidden" value="{{ dag.dag_id }}">
+            <input name="task_id" type="hidden">
+            <input name="execution_date" type="hidden">
+            <input name="origin" type="hidden" value="{{ request.base_url }}">
+            <div class="row">
+              <span class="btn-group col-md-9" data-toggle="buttons">
+                <label class="btn btn-default">
+                  <input type="checkbox" value="true" name="success_past" autocomplete="off">
+                  Past
+                </label>
+                <label class="btn btn-default">
+                  <input type="checkbox" value="true" name="success_future" autocomplete="off">
+                  Future
+                </label>
+                <label class="btn btn-default">
+                  <input type="checkbox" value="true" name="success_upstream" autocomplete="off">
+                  Upstream
+                </label>
+                <label class="btn btn-default">
+                  <input type="checkbox" value="true" name="success_downstream" autocomplete="off">
+                  Downstream
+                </label>
+              </span>
+              <span class="col-md-3">
+                <button id="btn_success" type="submit" class="btn btn-primary btn-block">
+                  Mark Success
+                </button>
+              </span>
+            </div>
           </form>
           <hr/>
           <span class="btn-group" id="extra_links"></span>
         </div>
-
         <div class="modal-footer">
-          <button type="button" class="btn btn-default" data-dismiss="modal">
-            Close
-          </button>
+          <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
         </div>
       </div>
     </div>
   </div>
   <!-- Modal for dag -->
-  <div class="modal fade" id="dagModal"
-        tabindex="-1" role="dialog"
-      aria-labelledby="dagModalLabel" aria-hidden="true">
+  <div class="modal fade" id="dagModal" tabindex="-1" role="dialog" aria-labelledby="dagModalLabel" aria-hidden="true">
     <div class="modal-dialog">
       <div class="modal-content">
         <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
           <h4 class="modal-title" id="dagModalLabel">
-            <span id='dag_id'></span>
+            <span class="text-muted">DAG:</span> {{dag.dag_id}}
```

Comment:
I need to make the "DAG" text dynamic as this modal can be used for "SUBDAG" too.